### PR TITLE
Add Glass Berry for collab use

### LINKED ISSUE #53: Importing the Glass Berry from Berry Helper
You know the deal, it's a berry which shatters if you dash while holding it.

I've got the code and everything. I just need to polish it and bring it in. I don't even know if anyone wants to use it, but I'm making it available, dammit.

### PR DESCRIPTION
**Pull request by GreyMario**
Initiating a dash while carrying this berry breaks it. Otherwise, it acts as a good old Red Strawberry, tracked by all berry-tracking features and collectable in the normal manner of "landing in a safe place for nine frames".

Closes #53

### CODE REVIEW

File: Entities/GlassBerry.cs [hunk 1]
```diff
@@ -0,0 +1,376 @@
...
+    // It will appear in the pause menu tracker, as well as all places where red berries are tallied.
+    // This custom Strawberry does not "block collection".
+    // It has normal collection rules, so it can be part of the berry train chain.
+    [RegisterStrawberry(true, false)]
+    [CustomEntity("SpringCollab2020/glassBerry")]
+    class GlassBerry : Entity, IStrawberry, IStrawberrySeeded {
+        // Requested implementations for using IStrawberrySeeded. 
+        // These are the most basic implementations thereof and there is likely no reason to change these.
+        public List<GenericStrawberrySeed> Seeds { get; }
+        public string gotSeedFlag { get { return "collected_seeds_of_" + ID.ToString(); } }
```
Comment (0x0ade) [inline Entities/GlassBerry.cs]:
```suggestion
        public string gotSeedFlag => "collected_seeds_of_" + ID.ToString();
```

File: Entities/GlassBerry.cs [hunk 2]
```diff
@@ -0,0 +1,376 @@
...
+            Collider = new Hitbox(14f, 14f, -7f, -7f);
+            Add(new PlayerCollider(new Action<Player>(OnPlayer), null, null));
+            Add(new MirrorReflection());
+            Add(Follower = new Follower(ID, null, new Action(OnLoseLeader)));
+            Follower.FollowDelay = 0.3f;
+
+            // As part of the showcase, the Silver Berry is compatible with the GenericStrawberrySeed.
+            // It doesn't make sense for a Silver Berry to have seeds, so this is purely for demonstration.
+            bool flag = data.Nodes != null && data.Nodes.Length != 0;
+            if (flag) {
```
Comment (0x0ade) [inline Entities/GlassBerry.cs]: `flag` isn't needed outside of the if statement condition, can be inlined.

File: Entities/GlassBerry.cs [hunk 3]
```diff
@@ -0,0 +1,376 @@
...
+            Add(sprite);
+
+            if (!isOwned)
+                sprite.Play("idle");
+            else
+                sprite.Play("idleGhost");
+
+            // Strawberries have certain special effects during their animation sequence.
+            // This adds a handler to enable this.
+            sprite.OnFrameChange = new Action<string>(OnAnimate);
```
Comment (0x0ade) [inline Entities/GlassBerry.cs]: Is `new Action<string>` necessary?

File: Entities/GlassBerry.cs [hunk 4]
```diff
@@ -0,0 +1,376 @@
...
+
+        // Basic Strawberry setup. This usually won't change too much.
+        public GlassBerry(EntityData data, Vector2 offset, EntityID gid) {
+            ID = gid;
+            Position = (start = data.Position + offset);
+
+            isOwned = SaveData.Instance.CheckStrawberry(ID);
+            Depth = -100;
+            Collider = new Hitbox(14f, 14f, -7f, -7f);
+            Add(new PlayerCollider(new Action<Player>(OnPlayer), null, null));
```
Comment (0x0ade) [inline Entities/GlassBerry.cs]: Are `new Action<Player>` and the additional `null`s necessary?

File: Entities/GlassBerry.cs [hunk 5]
```diff
@@ -0,0 +1,376 @@
...
+            Add(new MirrorReflection());
+            Add(Follower = new Follower(ID, null, new Action(OnLoseLeader)));
```
Comment (0x0ade) [inline Entities/GlassBerry.cs]: Is `new Action` necessary?

File: Entities/GlassBerry.cs [hunk 6]
```diff
@@ -0,0 +1,376 @@
...
+                bloom.Visible = light.Visible = false;
+            }
+
+            // Let's be polite and turn down the bloom a little bit if the base level has bloom.
+            if (SceneAs<Level>().Session.BloomBaseAdd > 0.1f)
+                bloom.Alpha *= 0.5f;
+
+            // The Glass Berry's gimmick is that it breaks if the player dashes while carrying it.
+            // So we need a generic DashListener.
+            Add(new DashListener { OnDash = new Action<Vector2>(OnDash) });
```
Comment (0x0ade) [inline Entities/GlassBerry.cs]: Is `new Action<Vector2>` necessary?

File: Entities/GlassBerry.cs [hunk 7]
```diff
@@ -0,0 +1,376 @@
...
+            Audio.Play("event:/game/general/diamond_touch", Position);
+            yield return 0.05f;
+
+
+            Input.Rumble(RumbleStrength.Medium, RumbleLength.Medium);
+
+            // Add obvious "shard" lines
+            for (int i = 0; i < 12; i++) {
+                float dir = Calc.Random.NextFloat(6.2831855f);
+                base.SceneAs<Level>().ParticlesFG.Emit(StrawberrySeed.P_Burst, 1, Position + Calc.AngleToVector(dir, 4f), Vector2.Zero, dir);
```
Comment (0x0ade) [inline Entities/GlassBerry.cs]: This `base` access doesn't seem necessary.

File: Entities/GlassBerry.cs [hunk 8]
```diff
@@ -0,0 +1,376 @@
...
+            session.UpdateLevelStartDashes();
+
+            // Coroutines allow certain processes to run independently, so to speak.
+            Add(new Coroutine(CollectRoutine(collectIndex), true));
+        }
+
+        private IEnumerator CollectRoutine(int collectIndex) {
+            Level level = base.Scene as Level;
+            base.Tag = Tags.TransitionUpdate;
+            base.Depth = -2000010;
```
Comment (0x0ade) [inline Entities/GlassBerry.cs]: The `base` accesses on this and the preceding lines don't seem necessary. Also, `Scene as Level` isn't consistent with the previous `SceneAs<Level>`.

**Comment by GreyMario** [comment]
Wow, I swear I scrubbed this during literally transcribing the decompile... (except glassbreak, which I know I copied directly from the old berryhelper whoops)

I guess I do need to make another pass. 🐬💦

File: Entities/GlassBerry.cs (hunk 1 above)
Comment (GreyMario) [reply Entities/GlassBerry.cs]: This isn't something I really learned about yet; I'll make a note of actually learning the thing.

**Comment by GreyMario** [comment]
BREAKING NEWS: Broom shoved haphazardly through code, displacing vital semicolons and putting six closing parentheses out of home. More at six.

File: Ahorn/entities/glassBerry.jl [hunk 9]
```diff
@@ -0,0 +1,64 @@
+﻿module SpringCollab2020GlassBerryModule
+
+using ..Ahorn, Maple
+
+@mapdef Entity "SpringCollab2020/glassBerry" GlassBerry(x::Integer, y::Integer, checkpointID::Integer=-1, order::Integer=-1, nodes::Array{Tuple{Integer, Integer}, 1}=Tuple{Integer, Integer}[])
+
+const placements = Ahorn.PlacementDict(
+	"Strawberry (Glass) (Spring Collab 2020)" => Ahorn.EntityPlacement(
```
Comment (Cruor) [inline Ahorn/entities/glassBerry.jl]:
```suggestion
	"Glass Strawberry (Spring Collab 2020)" => Ahorn.EntityPlacement(
```

**Pull request merged**